skbio/sequence/_sequence.py

```diff
@@ -547,6 +547,122 @@ def observed_chars(self):
     def _string(self):
         return self._bytes.tostring()
 
+    @classmethod
+    @experimental(as_of="0.4.0-dev")
+    def concat(cls, seqs, how='strict'):
+        """ Concatenate an iterable of Sequence objects.
+
+        Parameters
+        ----------
+        seqs : iterable (Sequence)
+            An iterable of Sequence objects or appropriate subclasses.
+        how : {'strict', 'inner', 'outer'}, optional
+            How to intersect the `positional_metadata` of the sequences.
+            If 'strict': the `positional_metadata` must have the exact same
+            columns; 'inner': an inner-join of the columns; 'outer': an
+            outer-join of the columns.
+
+        Returns
+        -------
+        Sequence
+            The returned sequence will be an instance of the class which
+            called this class-method.
+
+        Raises
+        ------
+        ValueError
+            If `how` is not one of: 'strict', 'inner', or 'outer'.
+        ValueError
+            If `how` is 'strict' and the `positional_metadata` of each sequence
+            does not have the same cloumns.
+        TypeError
+            If the sequences cannot be cast as the calling class.
+
+        Examples
+        --------
+        A simple example:
+        >>> from skbio import DNA, Sequence
+        >>> s1 = DNA("ACGT")
+        >>> s2 = DNA("GGAA")
+        >>> DNA.concat([s1, s2])
+        DNA
+        -----------------------------
+        Stats:
+            length: 8
+            has gaps: False
+            has degenerates: False
+            has non-degenerates: True
+            GC-content: 50.00%
+        -----------------------------
+        0 ACGTGGAA
+
+        Type coercion:
+        >>> Sequence.concat([s1, s2])
+        Sequence
+        -------------
+        Stats:
+            length: 8
+        -------------
+        0 ACGTGGAA
+
+        Positional metadata is conserved:
+        >>> s1 = DNA('AcgT', lowercase='one')
+        >>> s2 = DNA('GGAA', lowercase='one',
+        ...          positional_metadata={'two': [1, 2, 3, 4]})
+        >>> DNA.concat([s1, s2], how='outer')
+        DNA
+        -----------------------------
+        Positional metadata:
+            'one': <dtype: bool>
+            'two': <dtype: float64>
+        Stats:
+            length: 8
+            has gaps: False
+            has degenerates: False
+            has non-degenerates: True
+            GC-content: 50.00%
+        -----------------------------
+        0 ACGTGGAA
+
+        """
+        seqs = list(seqs)
+        for seq in seqs:
+            seq._assert_can_cast_to(cls)
+
+        if how not in {'strict', 'inner', 'outer'}:
```

Comment:
recommend bumping this check to the top b/c it's the quickest (`seqs` could contain an unbounded number of sequences)